packages/react-native-web/src/exports/InteractionManager/immediateShim.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) Nicolas Gallagher.
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+// Globally Unique Immediate ID.
+let GUID = 1;
+
+// A global set of the currently cleared immediate id.
+const clearedIds: Set<number> = new Set();
+
+const scheduleMicrotask =
+  typeof window !== 'undefined' && typeof window.queueMicrotask === 'function'
+    ? window.queueMicrotask
+    : typeof Promise !== 'undefined'
+    ? (callback) =>
+        Promise.resolve(null)
+          .then(callback)
+          .catch((error) => {
+            setTimeout(() => {
+              throw error;
+            });
+          })
+    : setTimeout;
+
+/**
+ * Shim the setImmediate API on top of queueMicrotask.
+ * @param {function} callback Callback to be invoked before the end of the
+ * current JavaScript execution loop.
+ * @param args
+ */
+function setImmediate(callback: Function, ...args: any): number {
```

Comment:
Do we need to use setImmediate? It's not a web standard and I'm hesitant to introduce a shim if you think there could be other ways to implement the InteractionManager behavior

Reply:
Oh, sorry, the name here may be misleading. It is different from [`setImmediate` in MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate#notes).
The former actually tries to use `queueMicrotask` to create a micro task ([polyfill](https://github.com/facebook/react/blob/2655c9354d8e1c54ba888444220f63e836925caa/packages/react-noop-renderer/src/createReactNoop.js#L421-L433) is `Promise.resolve`/`setTimeout`),  while the latter is a macro task(similar to `setTimeout`, except that it gets executed immediately, [polyfill](https://github.com/YuzuJS/setImmediate) is `postMessage`/`MessageChannel`/`setTimeout`).
I thought it would be better to execute these tasks that have been deferred by `InteractionManager` before the next macro task.
Here is my current understanding, please correct me if there is a mistake 😂:
![image](https://user-images.githubusercontent.com/8579651/203298666-dd67d3cc-c709-4129-80e1-56653a38e3e9.png)

* `queueMicrotask/Promise` compatibility is about 97.68%
* `requestAnimationFrame` compatibility is about 98.48%
* native `requestIdleCallback` compatibility is about 	77.96%

Which way do you prefer? 😄
### Some references:
* > As can be seen from the following examples, the best way of thinking about [queueMicrotask()](https://html.spec.whatwg.org/multipage/timers-and-user-prompts.html#dom-queuemicrotask) is as a mechanism for rearranging synchronous code, effectively placing the queued code immediately after the currently executing synchronous JavaScript has run to completion.
* https://stackoverflow.com/questions/43050448/when-will-requestanimationframe-be-executed

Reply:
And actually, the shim file can be deleted to avoid misleading if needed.
```diff
diff --git a/packages/react-native-web/src/exports/InteractionManager/index.js b/packages/react-native-web/src/exports/InteractionManager/index.js
index e3b535ab..5ccc7caf 100644
--- a/packages/react-native-web/src/exports/InteractionManager/index.js
+++ b/packages/react-native-web/src/exports/InteractionManager/index.js
@@ -94,6 +94,20 @@ let _nextUpdateHandle: TimeoutID | number = 0;
 let _inc = 0;
 let _deadline = -1;

+const scheduleMicrotask =
+  typeof window !== 'undefined' && typeof window.queueMicrotask === 'function'
+    ? window.queueMicrotask
+    : typeof Promise !== 'undefined'
+      ? (callback) =>
+        Promise.resolve(null)
+          .then(callback)
+          .catch((error) => {
+            setTimeout(() => {
+              throw error;
+            });
+          })
+      : setTimeout;
+
 /**
  * Schedule an asynchronous update to the interaction state.
  */
@@ -102,7 +116,8 @@ function _scheduleUpdate() {
     if (_deadline > 0) {
       _nextUpdateHandle = setTimeout(_processUpdate);
     } else {
-      _nextUpdateHandle = setImmediate(_processUpdate);
+      scheduleMicrotask(_processUpdate);
+      _nextUpdateHandle = 1;
     }
   }
 }
```